manta-pay/src/crypto/hash/poseidon.rs

```diff
@@ -81,30 +128,30 @@ where
     derive(Deserialize, Serialize),
     serde(
         bound(
-            deserialize = "S::Field: Deserialize<'de>",
-            serialize = "S::Field: Serialize"
+            deserialize = "S::ParameterField: Deserialize<'de>",
+            serialize = "S::ParameterField: Serialize"
         ),
         crate = "manta_util::serde",
         deny_unknown_fields
     )
 )]
 #[derive(derivative::Derivative)]
 #[derivative(
-    Clone(bound = "S::Field: Clone"),
-    Debug(bound = "S::Field: Debug"),
-    Eq(bound = "S::Field: Eq"),
-    Hash(bound = "S::Field: Hash"),
-    PartialEq(bound = "S::Field: PartialEq")
+    Clone(bound = "S::ParameterField: Clone"),
+    Debug(bound = "S::ParameterField: Debug"),
+    Eq(bound = "S::ParameterField: Eq"),
+    Hash(bound = "S::ParameterField: Hash"),
+    PartialEq(bound = "S::ParameterField: PartialEq")
 )]
 pub struct Hasher<S, COM, const ARITY: usize>
 where
     S: Specification<COM>,
 {
     /// Additive Round Keys
-    additive_round_keys: Vec<S::Field>,
+    additive_round_keys: Vec<S::ParameterField>,
 
     /// MDS Matrix
-    mds_matrix: Vec<S::Field>,
+    mds_matrix: Vec<S::ParameterField>,
 }
 
 impl<S, COM, const ARITY: usize> Hasher<S, COM, ARITY>
```

Comment:
I remember in original implementation by @bhgomes, the ARITY is associated constant? We might condier doing so as well?

Reply:
Do you mean in the `pub struct Hasher`? It seems there are only additive_round_keys and mds_matrix in the original implementation by @bhgomes 
See https://github.com/Manta-Network/manta-rs/blob/main/manta-pay/src/crypto/hash/poseidon.rs#L99-L108

Reply:
yes